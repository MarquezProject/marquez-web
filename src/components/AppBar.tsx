import React, { ReactElement } from 'react'
import { Link } from 'react-router-dom'
import { AppBar, Toolbar, Typography } from '@material-ui/core'
import {
  withStyles,
  createStyles,
  WithStyles as IWithStyles,
  Theme as ITheme
} from '@material-ui/core/styles'
import Box from '@material-ui/core/Box'
import 'typeface-rajdhani'
import '../img/we-logo.png'

const styles = ({ spacing, zIndex }: ITheme) => {
  return createStyles({
    appBar: {
      zIndex: zIndex.drawer + 1
    },
    rightToolbar: {
      marginLeft: 'auto',
      marginRight: 2.5
    },
    feedbackButton: {
      padding: spacing(1),
      backgroundColor: '#2B2B33',
      color: 'white',
      fontSize: 15,
      borderWidth: 1.2,
      '&:hover': {
        color: '#71ddbf',
        cursor: 'pointer',
        borderColor: '#71ddbf'
      }
    }
  })
}

interface IProps extends IWithStyles<typeof styles> {}

const MyAppBar = (props: IProps): ReactElement => {
  const { classes } = props
  return (
    <AppBar position='fixed' className={classes.appBar}>
      <Toolbar>
        <Box mr={2}>
          <img src='img/we-logo.png' height={50} alt='WeWork Logo' />
        </Box>
        <Typography
          variant='h4'
          color='inherit'
          noWrap
          style={{ fontFamily: 'rajdhani', fontWeight: 'bold' }}
        >
          <Link to='/' className='link'>
            MARQUEZ
          </Link>
        </Typography>
        <div className={classes.rightToolbar}>
          <a href='https://forms.gle/gWj8fn8iYiTsKEPD6' target='_blank' rel='noopener noreferrer'>
            <button className={classes.feedbackButton}> Feedback </button>
          </a>
        </div>
      </Toolbar>
    </AppBar>
  )
}

export default withStyles(styles)(MyAppBar)
