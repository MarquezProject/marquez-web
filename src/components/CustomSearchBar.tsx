import React, { FunctionComponent, useState } from 'react'
import SearchBar from 'material-ui-search-bar'
import { findMatchingEntities } from '../actionCreators'
import {
  withStyles,
  createStyles,
  WithStyles as IWithStyles,
  Theme as ITheme
} from '@material-ui/core/styles'

import { useHistory } from "react-router-dom";

interface IProps {
  findMatchingEntities: typeof findMatchingEntities
  setShowJobs: (bool: boolean) => void
  showJobs: boolean
}

type IAllProps = IWithStyles<typeof styles> & IProps

const styles = (_theme: ITheme) => {
  return createStyles({
    search: {
      position: 'fixed',
      bottom: '52vh',
      width: '90%',
      left: '5%',
      zIndex: 4
    }
  })
}

const CustomSearchBar: FunctionComponent<IAllProps> = props => {
  const [search, setSearch] = useState("")
  const { classes } = props
  let history = useHistory();

  const searchChanged = (searchString: string) => {
    setSearch(searchString)
    props.findMatchingEntities(searchString)
    searchString == '' ? props.setShowJobs(false) : props.setShowJobs(true)
  }

  const cancelledSearch = () => {
    setSearch('')
    searchChanged('')
  }

  const onRequestSearch = () => {
    history.push("/");
  }

  return (
    <SearchBar
      className={classes.search}
      value={search}
      onChange={searchChanged}
      onCancelSearch={cancelledSearch}
      onRequestSearch={onRequestSearch}
    />
  )
}


export default withStyles(styles)(CustomSearchBar)
