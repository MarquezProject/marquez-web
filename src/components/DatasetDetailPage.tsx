import React, { FunctionComponent } from 'react'
import {
  withStyles,
  createStyles,
  WithStyles as IWithStyles
} from '@material-ui/core/styles'
import { Typography, Box, Tooltip, Fab, Table, TableCell, TableHead, TableRow, TableBody, Paper } from '@material-ui/core'

import CloseIcon from '@material-ui/icons/Close'

import tagToBadge from '../config/tag-to-badge'
import InfoIcon from '@material-ui/icons/Info'

import { formatUpdatedAt } from '../helpers'

import { useParams, useHistory } from 'react-router-dom'
import _find from 'lodash/find'
import _keys from 'lodash/keys'

import { IDataset } from '../types'

const styles = () => {
  return createStyles({
    root: {
      marginTop: '52vh',
      height: '48vh',
      padding: '0 6% 1%',
    },
    tagContainer: {
      display: 'flex',
      margin: '12px 12px 0px 0px'
    },
    noData: {
      padding: '125px 0 0 0'
    },
    infoIcon: {
      paddingLeft: '3px',
      paddingTop: '3px'
    },
    tableCell: {
      display: 'flex',
      paddingTop: '12px',
      flexFlow: 'row nowrap',
      flexGrow: 1,
      flexBasis: 0
    },
    tableRow: {
      display: 'flex',
      width: '100%'
    },
    paper: {
      overflowX: 'auto',
      marginTop: '10px',
      display: 'flex',
      flexFlow: 'column nowrap'
    },
    updated: {
      marginTop: '10px'
    },
    closeButton: {
      color: '#7D7D7D',
      backgroundColor: '#ffffff'
    },
    tagHolder: {
      display: 'flex',
      padding: '9px 12px'
    }
  })
}

type IProps = IWithStyles<typeof styles> & { datasets: IDataset[] }

const DatasetDetailPage: FunctionComponent<IProps> = props => {
  const { datasets, classes } = props
  const {
    root, paper, updated, tagContainer, noData, infoIcon, tableCell, tableRow, closeButton, tagHolder
  } = classes
  const { datasetName } = useParams()
  const history = useHistory()
  const dataset = _find(datasets, d => d.name === datasetName)
  if (!dataset) {
    return (
      <Box
        mt={10}
        display='flex'
        justifyContent="center"
        className={root}
      >
        <Typography align='center' className={noData}>
          No dataset by the name of <strong>&quot;{datasetName}&quot;</strong> found
        </Typography>
      </Box>
    )
  } else {
    const {
      name,
      description,
      tags = [],
      updatedAt,
      fields
    } = dataset

    return (
      <Box mt={10} className={root}>
        <Box display='flex' justifyContent='space-between'>
          <div>
            <Typography color='secondary' align='left'>
              <strong>{name}</strong>
            </Typography>
            <Typography color='primary' align='left'>
              {description}
            </Typography>
          </div>
          <div id='tagContainer' className={tagContainer}>
            <div className={tagHolder}>
              {_keys(tagToBadge.default).map((key: string) => {
                return (
                  <div key={key}>
                    <Tooltip className="tagWrapper" title={key} placement="top">
                      {tags.includes(key) ? tagToBadge.highlighted[key] : tagToBadge.default[key]}
                    </Tooltip>
                  </div>
                )
              })}
            </div>
            <Fab className={closeButton} onClick={() => history.push('/')} size="small" aria-label="edit">
              <CloseIcon />
            </Fab>
          </div>
        </Box>
        <Paper className={paper}>
          <Table size="small">
            <TableHead>
              <TableRow className={tableRow}>
                {
                  fields.map((field) => {
                    return (
                    <TableCell className={tableCell} key={field.name} align="center"><strong>{field.name}</strong>
                      <Tooltip title={field.type} placement="top">
                        <div className={infoIcon}>
                          <InfoIcon color='disabled' fontSize='small' />
                        </div>
                      </Tooltip>
                    </TableCell>
                    )
                  })
                }
              </TableRow>
            </TableHead>
            <TableBody>
              <TableRow className={tableRow}>
                {fields.map((field) => {
                  return <TableCell className={tableCell} key={field.name} align="left">{field.description || 'no description'}</TableCell>
                })}
              </TableRow>
            </TableBody>
          </Table>
        </Paper>
        <Typography className={updated} color='primary' align='right'>
          last updated: {formatUpdatedAt(updatedAt)}
        </Typography>
      </Box>
    )
  }
}

export default withStyles(styles)(DatasetDetailPage)
