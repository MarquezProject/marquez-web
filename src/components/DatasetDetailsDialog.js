import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';

import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import Button from '@material-ui/core/Button';

import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';

const styles = theme => ({
    root: {
      width: '100%',
      marginTop: theme.spacing.unit * 3,
      overflowX: 'auto',
    },
    table: {
      minWidth: 700,
    },
  });

function DatasetDetailsTable(props) {
    const { classes } = props;
    return (
        <Table>
            <TableBody>
                <TableRow>
                    <TableCell><b>Name</b></TableCell>
                    <TableCell>{props.details.name}</TableCell>
                </TableRow>
                <TableRow>
                    <TableCell><b>CreatedAt</b></TableCell>
                    <TableCell>{props.details.createdAt}</TableCell>
                </TableRow>
                <TableRow>
                    <TableCell><b>URN</b></TableCell>
                    <TableCell>{props.details.urn}</TableCell>
                </TableRow>
                <TableRow>
                    <TableCell><b>DatasourceURN</b></TableCell>
                    <TableCell>{props.details.datasourceUrn}</TableCell>
                </TableRow>
                <TableRow>
                    <TableCell><b>Description</b></TableCell>
                    <TableCell>{props.details.description}</TableCell>
                </TableRow>
            </TableBody>
        </Table>
    )
}

function DatasetDetailsDialog(props) {
    return (
        <Dialog
            fullWidth="sm"
            maxWidth="sm"
            open={props.open}
            onClose={props.onClose}
            aria-labelledby="max-width-dialog-title"
        >
            <DialogTitle id="max-width-dialog-title">Dataset Details</DialogTitle>
            <DialogContent>
                <DatasetDetailsTable details={props.datasetDetails} />
            </DialogContent>
            <DialogActions>
                <Button onClick={props.onClose} color="primary">
                    Close
                </Button>
            </DialogActions>
        </Dialog>
    );
}

DatasetDetailsDialog.propTypes = {
    open: PropTypes.bool.isRequired,
    onClose: PropTypes.func.isRequired,
    datasetDetails: PropTypes.object.isRequired
}

DatasetDetailsDialog.defaultProps = {
    open: false,
    onClose: function(){},
    jobDetails: null
}

export default withStyles(styles)(DatasetDetailsDialog);
