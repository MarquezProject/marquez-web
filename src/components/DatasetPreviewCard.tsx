import React, { ReactElement } from 'react'
import {
  withStyles,
  createStyles,
  WithStyles as IWithStyles,
  Theme as ITheme
} from '@material-ui/core/styles'
import { Typography, Box } from '@material-ui/core'
import { formatUpdatedAt } from '../helpers'
import tagToBadge from '../config/tag-to-badge'

import { IDatasetAPI } from '../types/api'

const styles = ({ palette }: ITheme) => {
  return createStyles({
    rightCol: {
      textAlign: 'right'
    },
    lastUpdated: {
      color: palette.grey[600]
    }
  })
}

const StyledTypography = withStyles({
  root: {
    maxWidth: '90%'
  }
})(Typography)

type IProps = IWithStyles<typeof styles> &
  Pick<IDatasetAPI, 'name' | 'description' | 'updatedAt' | 'tags'>
interface IState {}

class DatasetPreviewCard extends React.Component<IProps, IState> {
  render(): ReactElement {
    const { classes, name, description, updatedAt = 'error', tags = ['is_pii'] } = this.props
    return (
      <Box p={2} m={1} bgcolor='white' boxShadow={3} display='flex' justifyContent='space-between'>
        <div>
          <Typography color='secondary' variant='h3'>
            {name}
          </Typography>
          <StyledTypography color='primary'>{description}</StyledTypography>
        </div>
        <Box
          className={classes.rightCol}
          display='flex'
          flexDirection='column'
          alignItems='flex-end'
          justifyContent='space-between'
        >
          <div id='tagContainer'>{tags.map(t => tagToBadge[t])}</div>
          <Typography className={classes.lastUpdated}>{formatUpdatedAt(updatedAt)}</Typography>
        </Box>
      </Box>
    )
  }
}

export default withStyles(styles)(DatasetPreviewCard)
