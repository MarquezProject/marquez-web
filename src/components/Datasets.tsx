import React, { ReactElement } from 'react'
import * as RRD from 'react-router-dom'
import { Typography, Box } from '@material-ui/core'

interface IProps {}

interface IState {}

type IAllProps = RRD.RouteComponentProps & IProps

class Datasets extends React.Component<IAllProps, IState> {
  render(): ReactElement {
    return (
      <Box mt={10}>
        <Typography color='primary' align='center'>
          {' '}
          Datasets Component{' '}
        </Typography>
      </Box>
    )
  }
}

export default Datasets
