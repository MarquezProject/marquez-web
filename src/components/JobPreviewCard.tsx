import React, { ReactElement } from 'react'
import {
  withStyles,
  createStyles,
  WithStyles as IWithStyles,
  Theme as ITheme
} from '@material-ui/core/styles'
import { Link } from 'react-router-dom'

const globalStyles = require('../global_styles.css')
const { vibrantGreen } = globalStyles
import { Typography, Box, Tooltip } from '@material-ui/core'
import { formatUpdatedAt } from '../helpers'

import { IJobAPI } from '../types/api'

const { jobRunNew, jobRunFailed, jobRunCompleted, jobRunAborted, jobRunRunning } = globalStyles

const colorMap = {
  NEW: jobRunNew,
  FAILED: jobRunFailed,
  COMPLETED: jobRunCompleted,
  ABORTED: jobRunAborted,
  RUNNING: jobRunRunning
}

const styles = ({ palette, spacing }: ITheme) => {
  return createStyles({
    rightCol: {
      textAlign: 'right'
    },
    lastUpdated: {
      color: palette.grey[600]
    },
    status: {
      width: spacing(2),
      height: spacing(2),
      borderRadius: '50%'
    },
    failed: {
      backgroundColor: palette.error.main
    },
    passed: {
      backgroundColor: vibrantGreen
    },
    link: {
      textDecoration: 'none'
    }
  })
}

type IProps = IWithStyles<typeof styles> &
  Pick<IJobAPI, 'name' | 'description' | 'updatedAt' | 'latestRun'>
interface IState {}

const StyledTypography = withStyles({
  root: {
    maxWidth: '90%'
  }
})(Typography)

class JobPreviewCard extends React.Component<IProps, IState> {
  render(): ReactElement {
    const { classes, name, description, updatedAt = '', latestRun } = this.props

    return (
      <Link to={`/jobs/${name}`} className={classes.link}>
        <Box
          p={2}
          m={1}
          bgcolor='white'
          boxShadow={3}
          display='flex'
          justifyContent='space-between'
        >
          <div>
            <Typography color='secondary' variant='h3'>
              {name}
            </Typography>
            <StyledTypography color='primary'>{description}</StyledTypography>
          </div>
          <Box
            className={classes.rightCol}
            display='flex'
            flexDirection='column'
            alignItems='flex-end'
            justifyContent='space-between'
          >
            <Tooltip className="tagWrapper" title={latestRun.runState} placement="top">
              <div className={classes.status} style={{backgroundColor: colorMap[latestRun.runState]}} />
            </Tooltip>
            <Typography className={classes.lastUpdated}>{formatUpdatedAt(updatedAt)}</Typography>
          </Box>
        </Box>
      </Link>
    )
  }
}

export default withStyles(styles)(JobPreviewCard)
