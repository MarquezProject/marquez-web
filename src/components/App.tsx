import React, { ReactElement } from 'react'
import { Helmet } from 'react-helmet'
import { CssBaseline } from '@material-ui/core'
import AppBar from './AppBar'
const globalStyles = require('../global_styles.css')
const { neptune, telescopeBlack } = globalStyles
import { Route, Switch } from 'react-router-dom'
import { Grid } from '@material-ui/core'
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles'
import { createStore, applyMiddleware } from 'redux'
import { Provider } from 'react-redux'
import logger from 'redux-logger'
import createSagaMiddleware from 'redux-saga'
import { composeWithDevTools } from 'redux-devtools-extension'
import { createBrowserHistory } from 'history'
import { routerMiddleware, ConnectedRouter } from 'connected-react-router'
import {
  withStyles,
  createStyles,
  WithStyles as IWithStyles,
  Theme as ITheme
} from '@material-ui/core/styles'

import createRootReducer from '../reducers'
import rootSaga from '../sagas'
import HomeContainer from '../containers/HomeContainer'
import Toast from '../containers/ToastContainer'
import NetworkGraphContainer from '../containers/NetworkGraphContainer'
import DatasetsContainer from '../containers/DatasetsContainer'

const sagaMiddleware = createSagaMiddleware({
  onError: (error, _sagaStackIgnored) => {
    console.log('There was an error in the saga', error)
  }
})
const history = createBrowserHistory()
const historyMiddleware = routerMiddleware(history)

const middleware = __DEVELOPMENT__
  ? [sagaMiddleware, historyMiddleware, logger]
  : [sagaMiddleware, historyMiddleware]

const store = createStore(
  createRootReducer(history),
  composeWithDevTools(applyMiddleware(...middleware))
)

sagaMiddleware.run(rootSaga)
// see the defaults here for MUI Theme -- https://material-ui.com/customization/default-theme/?expend-path=$.typography
const theme = createMuiTheme({
  typography: {
    h3: {
      fontSize: '1rem',
      fontWeight: 700,
      lineHeight: 2
    },
    fontSize: 14
  },
  palette: {
    primary: {
      main: telescopeBlack
    },
    secondary: {
      main: neptune
    }
  }
})

const styles = (_theme: ITheme) => {
  return createStyles({
    root: {
      height: '100vh',
      display: 'flex'
    }
  })
}

interface IProps extends IWithStyles<typeof styles> {}

const TITLE = 'Marquez | Data Kit'

const App = ({ classes }: IProps): ReactElement => {
  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <MuiThemeProvider theme={theme}>
          <Helmet>
            <title>{TITLE}</title>
          </Helmet>
          <CssBaseline />
          <Grid direction='column' alignItems='stretch' classes={classes} justify='space-between'>
            <AppBar />
            <NetworkGraphContainer />
            <Switch>
              <Route path='/' exact component={HomeContainer} />
              <Route path='/datasets/:name' exact component={DatasetsContainer} />
            </Switch>
            <Toast />
          </Grid>
        </MuiThemeProvider>
      </ConnectedRouter>
    </Provider>
  )
}

export default withStyles(styles)(App)
