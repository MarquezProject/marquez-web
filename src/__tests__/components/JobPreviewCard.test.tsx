import { shallow } from 'enzyme'
import * as React from 'react'

import JobPreviewCard from '../../components/JobPreviewCard'
import { formatUpdatedAt } from '../../components/DatasetPreviewCard'

const jobs = require('../../../docker/db/data/jobs.json')

describe('formatUpdated Function', () => {
  const updatedAt = ''
  const formattedDate = formatUpdatedAt(updatedAt)
  it('Should return an empty string when passed a falsey value', () => {
    expect(formattedDate).toBe('')
  })
})

describe('JobPreviewCard Component', () => {
  const wrapper = shallow(<JobPreviewCard />)
  it('Should render', () => {
    expect(wrapper.exists()).toBe(true)
  })

  const job = jobs[0]

  wrapper.setProps(job)
  const componentText = wrapper.render().text()

  it('should render the job name', () => {
    expect(componentText).toContain(job.name)
  })
  it('should render the job description', () => {
    expect(componentText).toContain(job.description)
  })
  it('should render the job time', () => {
    expect(componentText).toContain(formatUpdatedAt(job.updatedAt))
  })
  test.skip('should render the job status', () => {
    expect(componentText).toContain(job.status)
  })
  it('renders a snapshot that matches previous', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
