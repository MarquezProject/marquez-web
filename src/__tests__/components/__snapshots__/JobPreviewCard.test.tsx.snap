// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobPreviewCard Component renders a snapshot that matches previous 1`] = `
<JobPreviewCard
  classes={
    Object {
      "failed": "JobPreviewCard-failed-4",
      "lastUpdated": "JobPreviewCard-lastUpdated-2",
      "passed": "JobPreviewCard-passed-5",
      "rightCol": "JobPreviewCard-rightCol-1",
      "status": "JobPreviewCard-status-3",
    }
  }
  description="calculates all possible dimensions from input geometry"
  inputs={
    Array [
      "galaxy.geometry",
    ]
  }
  location="https://github.com/Galaxy/blob/ecfc04de65137bfd25a0416d227eaba5325c8c63/calculate.galaxy_dimensions.sql"
  name="calculate.galaxy_dimensions"
  namespace="galactic-data"
  outputs={
    Array [
      "galaxy.dimensions",
    ]
  }
  type="BATCH"
/>
`;
